sdk/storage/azure-storage-datamovement/CMakeLists.txt

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required (VERSION 3.13)
+project(azure-storage-datamovement LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+option(FETCH_SOURCE_DEPS "build source dependencies" OFF)
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake-modules")
+
+include(AzureVcpkg)
+include(AzureVersion)
+include(AzureCodeCoverage)
+include(AzureTransportAdapters)
+include(AzureDoxygen)
+include(AzureGlobalCompileOptions)
+include(AzureConfigRTTI)
+include(AzureBuildTargetForCI)
+
+az_vcpkg_integrate()
```

Comment:
I don't know how it works, and I don't know the difference. But I fixed all storage packages in this PR https://github.com/Azure/azure-sdk-for-cpp/pull/3604.

Reply:
> I don't know how it works

when we call `az_vcpkg_integrate()`, we check if the env var `VCPKG_ROOT` is set. If yes, we use it to set the CMAKE_TOOLCHAIN_FILE to Vcpkg folder.

But it turns out that we need to set this before calling `project(..)`, otherwise it has no effect and the toolchain file is not automatically set.